src/gui/accountsettings.cpp

```diff
@@ -333,6 +334,43 @@ void AccountSettings::slotEditCurrentIgnoredFiles()
     openIgnoredFilesDialog(f->path());
 }
 
+void AccountSettings::slotOpenMkFolderDialog()
+{
+    const auto selected = _ui->_folderList->selectionModel()->currentIndex();
+
+    if (!selected.isValid()) {
+        qCWarning(lcAccountSettings) << "Selection model current folder index is not valid.";
+        return;
+    }
+
+    const auto classification = _model->classify(selected);
+
+    if (classification != FolderStatusModel::SubFolder && classification != FolderStatusModel::RootFolder) {
+        return;
+    }
+
+    QString fileName;
+
+    if (classification == FolderStatusModel::RootFolder) {
```

Comment:
Alternatively I guess we could have added support for FolderPathRole also on the RootFolder? This way we wouldn't need that if in the first place?

Reply:
@er-vin Hmmmmmm, I don't know. This will introduce significantly more work and more QA + more chances for an error. Do we really need this sacrifice for the sake of not having `if/else`? I mean, if there were no other way, or if we've happened to become in need of using the same condition somewhere else, then I'd certainly vote for such refactoring, but, now, I see more evil than good if I do this.

Reply:
Fair enough, let's not do this. It came mostly out of my sheer surprise that this role wasn't supported on said roots... feels like it was overlooked back then.